centreon/www/class/centreonWidget.class.php

```diff
@@ -957,17 +963,92 @@ public function getWidgetPreferences($widgetId)
     {
-        $query = 'UPDATE widgets '
-            . 'SET title = :title '
-            . 'WHERE widget_id = :widgetId';
-        $stmt = $this->db->prepare($query);
-        $stmt->bindParam(':title', $newName, PDO::PARAM_STR);
-        $stmt->bindParam(':widgetId', $widgetId, PDO::PARAM_INT);
-        $stmt->execute();
+        global $centreon;
+        // ACL check
+        if ($centreon->user->admin === '1') {
+            $updateSql = <<<'SQL'
+                    UPDATE widgets
+                    SET title = :title
+                    WHERE widget_id = :widgetId
+                SQL;
+        } else {
+            $goupIds = $this->userGroups !== [] ? implode(',', $this->userGroups) : '';
+            $groupClause = $goupIds ? " OR cvur.usergroup_id IN ({$goupIds})" : '';
+            $updateSql = <<<'SQL'
+                    UPDATE widgets AS w
+                    INNER JOIN widget_views AS wv
+                        ON wv.widget_id = w.widget_id
+                    SET w.title = :title
+                    WHERE w.widget_id = :widgetId
+                    AND (
+                            /* Case 1: user is owner of the parent custom view */
+                            EXISTS (
+                                SELECT 1
+                                FROM custom_view_user_relation AS cvur
+                                WHERE cvur.custom_view_id = wv.custom_view_id
+                                AND cvur.user_id = :userId
+                                AND cvur.is_owner = 1
+                            )
+                            OR
+                            /* Case 2: (user OR any group) AND not locked */
+                            EXISTS (
+                                SELECT 1
+                                FROM custom_view_user_relation AS cvur
+                                WHERE cvur.custom_view_id = wv.custom_view_id
+                                AND cvur.locked = 0
+                                AND (
+                                        cvur.user_id = :userId
+                                        {$groupClause}
+                                    )
+                            )
+                        )
+                SQL;
+        }
+        try {
+            $params = [
+                QueryParameter::int('widgetId', $widgetId),
+                QueryParameter::string('title', $newName),
+            ];
+            if ($centreon->user->admin !== '1') {
+                $params[] = QueryParameter::int('userId', $this->userId);
+            }
+            $queryParameters = QueryParameters::create($params);
+            $nbAffectedRows = $this->db->update($updateSql, $queryParameters);
+
+            // If no row → no permission
+            if ($nbAffectedRows === 0) {
+                $msgError = 'Access denied to rename widget';
+                CentreonLog::create()->error(
+                    CentreonLog::TYPE_SQL,
```

Comment:
We don't use sql-error.log now.
```suggestion
                    CentreonLog::TYPE_BUSINESS_LOG,
```